src/extensions/scratch3_music/index.js

```diff
@@ -884,9 +993,20 @@ class Scratch3MusicBlocks {
 
         const engine = util.runtime.audioEngine;
         const chain = engine.createEffectChain();
-        chain.setEffectsFromTarget(util.target);
+        const soundEffects = Object.assign({}, util.target.soundEffects);
+        if (pitchShift) {
+            soundEffects.pitch = (soundEffects.pitch || 0) + pitchShift;
+        }
+        chain.setEffectsFromTarget(soundEffects);
         player.connect(chain);
 
+        // Dirty hack to make the pitch effect work - otherwise the effect isn't applied, for some reason.
+        // (Other audio effects work fine without this, though.)
+        const pitchEffect = chain._effects.find(effect => effect.name === 'pitch')
+        if (pitchEffect) {
+            pitchEffect.updatePlayer(player);
+        }
```

Comment:
This is a hack, of course, because it directly searches for `pitchEffect` by accessing `chain._effects`. I couldn't figure out any other way to make the effect apply correctly, though.